code-challenges.test.js

```diff
@@ -17,38 +25,68 @@
 
 // a) Create a test with an expect statement using the variables provided.
 
+describe('coded', () => {
+    test("returns a coded message", () => {
+        expect(coded(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(coded(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(coded(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
 var secretCodeWord1 = "Lackadaisical"
 // Expected output: "L4ck4d41s1c4l"
 var secretCodeWord2 = "Gobbledygook"
 // Expected output: "G0bbl3dyg00k"
 var secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
-
-
-
-
 // b) Create the function that makes the test pass.
 
-
-
-
+//create a function called coded 
+//passing in a string
+const coded = (string) => {
+//use .replace for 'a','e','i','o' to 4,3,1,0 and return
+   return string.replace(/a/g,'4').replace(/e|E/g, '3').replace(/i/g,'1').replace(/o/g,'0')
+}
+console.log(coded(secretCodeWord1))
+console.log(coded(secretCodeWord2))
+console.log(coded(secretCodeWord3))
```

Comment:
YES!!! You are the only one so far to figure out this problem with regex. Great job!

Reply:
Thank you